gobblin-service/src/main/java/org/apache/gobblin/service/modules/flowgraph/datanodes/hive/HiveDataNode.java

```diff
@@ -17,30 +17,33 @@
 
 package org.apache.gobblin.service.modules.flowgraph.datanodes.hive;
 
+
 import java.io.IOException;
 import java.net.URI;
 
 import com.google.common.base.Preconditions;
 import com.typesafe.config.Config;
 
+import java.util.Arrays;
 import joptsimple.internal.Strings;
 import lombok.EqualsAndHashCode;
 import lombok.Getter;
 
 import org.apache.gobblin.annotation.Alpha;
 import org.apache.gobblin.service.modules.flowgraph.FlowGraphConfigurationKeys;
-import org.apache.gobblin.service.modules.flowgraph.datanodes.fs.HdfsDataNode;
+import org.apache.gobblin.service.modules.flowgraph.datanodes.fs.FileSystemDataNode;
 import org.apache.gobblin.util.ConfigUtils;
 
 
 /**
- * An abstract {@link HiveDataNode} implementation. In addition to the required properties of a {@link HdfsDataNode}, an {@link HiveDataNode}
+ * An abstract {@link HiveDataNode} implementation. In addition to the required properties of a {@link FileSystemDataNode}, an {@link HiveDataNode}
  * must have a metastore URI specified.
  */
 @Alpha
 @EqualsAndHashCode (callSuper = true)
-public class HiveDataNode extends HdfsDataNode {
+public class HiveDataNode extends FileSystemDataNode {
   public static final String METASTORE_URI_KEY = FlowGraphConfigurationKeys.DATA_NODE_PREFIX + "hive.metastore.uri";
+  private static final String[] HIVE_SUPPORTED_SCHEME = {"adl", "abfs", "hdfs"};
```

Comment:
Maybe we should drop supported schemes check. There could be other schemes (e.g. Hive on AWS) which the current list misses.

Reply:
removed the check and related test cases